wagtail/admin/templates/wagtailadmin/userbar/base.html

```diff
@@ -22,13 +22,13 @@
                     <use href="#icon-wagtail-icon"></use>
                   </svg>
                 {% endblock %}
-                <span class="wagtail-userbar-help-text">{% trans 'Go to Wagtail admin interface' %}</span>
-            </div>
-            <div class='wagtail-userbar-items'>
+                <span class="wagtail-userbar-help-text">{% trans 'View Wagtail quick actions' %}</span>
+            </button>
+            <ul aria-labelledby="wagtail-userbar-trigger" class="wagtail-userbar-items" role="menu">
                 {% for item in items %}
                     {{ item|safe }}
                 {% endfor %}
-            </div>
+            </ul>
```

Comment:
@thibaudcolas The switch to `ul` and `li` elements is inspired by the [W3 WAI actions menu button example](https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-actions.html) (see html source code at the bottom of the page).

It is my understanding and rule of thumb that 'if it reads like a list, it's probably a list'. The menu is a list of items - so I think it should use list elements. You're right that we are currently using `role="presentation"` on the `li` elements, which essentially masks their list item role. I'm not exactly sure why I put it there in the first place. I'll check what happens when I remove `role="presentation"`.

As for the accidental inheritance of styles, is there something we can do to minimise that risk or is switching it back to `div`s a better option?

Reply:
Ah, got it. Since we’re overriding the `role` of both the `ul` and `li` anyway, as well as all the default list styles for those elements, I don’t think there is much of an advantage to using those tags over `div`. The one thing I could think of is if the CSS fails to load – then having the userbar look like a list would definitely be an improvement.

I’m happy for us to retain `ul` and `li` as long as we reduce the risk of styles clashes with the host page – this could be as simple as adding a:

```css
all: initial !important;
```

as the first style declaration for both the `ul` and the `li.

Reply:
```css
all: initial !important;
```

Tried this but breaks styles. Might just be easier to revert back to divs in that case...